sopel/plugins/rules.py

```diff
@@ -1141,53 +1150,59 @@ def is_threaded(self):
     def is_unblockable(self):
         return self._unblockable
 
-    def is_user_rate_limited(self, nick):
-        metrics: RuleMetrics = self._metrics_nick.get(nick, RuleMetrics())
-        now = datetime.datetime.utcnow()
-        rate_limit = datetime.timedelta(seconds=self._user_rate_limit)
-        return metrics.is_limited(now - rate_limit)
+    def get_user_metrics(self, nick) -> RuleMetrics:
```

Comment:
Let's go full annotations:

```suggestion
    def get_user_metrics(self, nick: Identifier) -> RuleMetrics:
```